packages/dustkit/src/rpc/getMessagePortRpcClient.ts

```diff
@@ -10,9 +10,14 @@ export const initMessage = "MessagePortRpcClient";
 export async function getMessagePortRpcClient(
   target: Window,
 ): Promise<SocketRpcClient<MessagePort>> {
+  let closed = false;
   return getSocketRpcClient({
     async getSocket({ onClose, onError, onOpen, onResponse }) {
-      let closed = false;
+      if (closed) {
+        // On closure, viem will try to reconnect, but we don't want that
```

Comment:
It gets called here: https://github.com/wevm/viem/blob/e50696f45e28b5206bef67f0ddc149b988a9e1bd/src/utils/rpc/socket.ts#L151

We dont want to set `reconnect` to false, cuz then it wouldnt call this on errors

This does lead to this error being throw in the client console everytime a chest is closed (doesnt break anything cuz the component unloads)